sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -361,17 +374,22 @@ private[spark] class SQLHistoryListener(conf: SparkConf, sparkUI: SparkUI)
 /**
  * Represent all necessary data for an execution that will be used in Web UI.
  */
-private[ui] class SQLExecutionUIData(
-    val executionId: Long,
-    val description: String,
-    val details: String,
-    val physicalPlanDescription: String,
-    val physicalPlanGraph: SparkPlanGraph,
-    val accumulatorMetrics: Map[Long, SQLPlanMetric],
-    val submissionTime: Long,
-    var completionTime: Option[Long] = None,
-    val jobs: mutable.HashMap[Long, JobExecutionStatus] = mutable.HashMap.empty,
-    val stages: mutable.ArrayBuffer[Int] = mutable.ArrayBuffer()) {
+private[ui] case class SQLExecutionUIData(
```

Comment:
nit: Since you move some fields out of the constructor, it should not be a case class. Most of the generated methods, e.g., `toString`, `hashCode`, `equals` won't work.